From 63ae789499395abc2b71fd46f57cac3c4ba1bd9d Mon Sep 17 00:00:00 2001
From: Vishnu Nair <vishnun@google.com>
Date: Wed, 8 Jan 2025 14:56:51 -0800
Subject: [PATCH] Check dump permissions before dumping --high-priority

Bug: 383328827
Test: new cts test
Flag: EXEMPT bug fix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:c617d697cdbef000cb416aefc08fc4a3cda4afa1)
Merged-In: Ie4b4e0001430924674b08eabf28c656a46527727
Change-Id: Ie4b4e0001430924674b08eabf28c656a46527727
---
 .../core/java/com/android/server/wm/WindowManagerService.java   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/wm/WindowManagerService.java b/services/core/java/com/android/server/wm/WindowManagerService.java
index 8e7c9cca88bb..930bfe71687c 100644
--- a/services/core/java/com/android/server/wm/WindowManagerService.java
+++ b/services/core/java/com/android/server/wm/WindowManagerService.java
@@ -7201,12 +7201,12 @@ public class WindowManagerService extends IWindowManager.Stub
 
     @Override
     public void dump(FileDescriptor fd, PrintWriter pw, String[] args) {
+        if (!DumpUtils.checkDumpPermission(mContext, TAG, pw)) return;
         PriorityDump.dump(mPriorityDumper, fd, pw, args);
     }
 
     @NeverCompile // Avoid size overhead of debugging code.
     private void doDump(FileDescriptor fd, PrintWriter pw, String[] args, boolean useProto) {
-        if (!DumpUtils.checkDumpPermission(mContext, TAG, pw)) return;
         boolean dumpAll = false;
 
         int opti = 0;
-- 
2.49.0.395.g12beb8f557-goog

